src/Compilers/CSharp/Portable/Symbols/NamespaceSymbol.cs

```diff
@@ -311,6 +311,23 @@ internal NamespaceSymbol GetNestedNamespace(NameSyntax name)
             return null;
         }
 
+        private ImmutableArray<NamedTypeSymbol> _lazyTypesWithExtensionMethods;
+        private ImmutableArray<NamedTypeSymbol> TypesWithExtensionMethods
```

Comment:
Consider renaming to TypesMightContainExtensionMethods. Also, consider if merged namespace should have its own implementation that wouldn't have to merge all types across all constituent parts in order to build the list.